src/chainlit/langchain/callbacks.py

```diff
@@ -500,6 +528,24 @@ async def on_llm_end(self, response: LLMResult, **kwargs: Any) -> None:
     async def on_chain_start(
         self, serialized: Dict[str, Any], inputs: Dict[str, Any], **kwargs: Any
     ) -> None:
+        path = serialized.get("kwargs", {}).get("llm", {}).get("id", [])
+        provider = ".".join(path)
+        provider_name = path[-1]
+        provider_path = ".".join(path[:-1])
+
+        if not any(provider == p.id for p in providers):
+            module = importlib.import_module(provider_path)
+            llm_class = getattr(module, provider_name)
```

Comment:
This is cool that we can automate this but it has limits:

1. We assume the env variables are available globally (where the llm could have been instantiated with the API key)
2. The inputs are not available

We could keep this approach as a last resort but also create a documentation on "How to Add a Langchain LLM provider" using the same LangchainGenericProvider class

Reply:
I think we should have the most automated approach by default. So that it's super smooth to integrate.
And have the "How to Add a Langchain LLM provider" option for more customized use cases, wdyt?